src/twisted/trial/test/test_tests.py

```diff
@@ -965,6 +965,28 @@ def test_multipleErrorsReported(self):
         self.assertEqual(error2.getErrorMessage(), "foo")
 
 
+    def test_cleanupsAreReentrant(self):
+        @self.test.addCleanup
+        def reenter():
+            self.test.append("foo")
+            self.test.addCleanup(self.test.append, "bar")
+
+        self.test.run(self.result)
+        self.assertEqual(["setUp", "runTest", "foo", "bar", "tearDown"], self.test.log)
+
+    def test_cleanupsIgnoreKeyboardInterrupt(self):
+        self.test.addCleanup(self.test.append, "foo")
+
+        @self.test.addCleanup
```

Comment:
For consistency, maybe just make a plain call instead of using the decoration syntax.

Reply:
I don't know what you mean, can you use `` ```suggest ``